csrc/transformer/inference/csrc/pt_binding.cpp

```diff
@@ -461,7 +461,7 @@ std::vector<at::Tensor> ds_softmax_context(at::Tensor& query_key_value,
     auto kv_cache = workspace + offset + (hidden_dim / heads) * (is_prompt ? 0 : soft_len - 1);
     size_t value_offset = bsz * Context::Instance().GetMaxTokenLenght() * hidden_dim;
 
-    T* temp_buf = (T*)output.data_ptr() + at::numel(output);
+    T* temp_buf = (T*)kv_cache+value_offset+value_offset;
```

Comment:
by using this part of the memory, we are corrupting the cache of the consecutive layer. May I suggest keeping the temp unchanged and change the workspace used for the attention-score [here ](https://github.com/andrewchernyh/DeepSpeed/blob/fix-gpt-long-generation/csrc/transformer/inference/csrc/pt_binding.cpp#L357)to always point at the last portion of the memory buffer? Note, that the main issue is that we are currently writing and reading at the same place when doing the last GeMM in this function and by moving the workspace to the end of the buffer, we can be sure of not corrupting the memory.

Reply:
you can do this, by just keeping the else part.

Reply:
Sure, will do it

Reply:
Fixed